test/e2e/specs/default-config/recordset/facet-within-facet.spec.ts

```diff
@@ -0,0 +1,740 @@
+              ],
+              select: [{
+                option: 3,
+                numRows: 1
+              }, {
+                option: 4,
+                numRows: 2,
+                numFacetFilters: 1
+              }]
+            },
+            {
+              index: 4,
+              name: popuFacetNames[4],
+              options: [
+                'All records with value', 'facet_2nd_layer_hub_inbound4_06', 'facet_2nd_layer_hub_inbound4_07',
+                'facet_2nd_layer_hub_inbound4_08', 'facet_2nd_layer_hub_inbound4_09'
+              ],
+              select: [{
+                option: 1,
+                numRows: 1,
+                numFacetFilters: 2
+              }],
+              popup: {
+                uiContext: rootUIContext + 'Entity path with shared prefix and filter:',
+                title: 'Select 2nd layer entity path with shared prefix and filter',
+                numRows: 4,
+                selectedChiclets: ['facet_2nd_layer_hub_inbound4_06']
+              }
+            }
+          ]
+        }
+      },
+      {
+        index: 5,
+        name: 'Select 2nd layer entity path with fast filter',
+        description: 'fast filter',
+        popup: {
+          uiContext: rootUIContext + ':',
+          title: 'Select Entity path with fast filter',
+          facetNames: popuFacetNames,
+          numRows: 8,
+          facets: [
+            {
+              index: 5,
+              name: popuFacetNames[5],
+              options: [
+                'All records with value', 'No value', 'facet_2nd_layer_hub_inbound5_13',
+                'facet_2nd_layer_hub_inbound5_14', 'facet_2nd_layer_hub_inbound5_15'
+              ],
+              select: [{
+                option: 1,
+                numRows: 5
+              }, {
+                option: 2,
+                numRows: 6
+              }],
+              popup: {
+                uiContext: rootUIContext + 'Entity path with fast filter:',
+                title: 'Select 2nd layer entity path with fast filter',
+                numRows: 3,
+                selectedChiclets: ['No value', 'facet_2nd_layer_hub_inbound5_13']
+              }
+            },
+            // open another one and make sure it works
+            {
+              index: 0,
+              name: popuFacetNames[0],
+              options: ['All records with value', 'facet_2nd_layer_hub_inbound1_13'],
+              popup: {
+                uiContext: rootUIContext + 'Entity path with fast filter:',
+                title: 'Select 2nd layer entity path',
+                numRows: 1,
+                selectedChiclets: []
+              }
+            }
+          ]
+        }
+      }
+    ]
+  }
+}
+
+/******************** helpers ************************/
+
+const getChaiseURL = (appName: string, tableName: string, baseURL: string | undefined, testInfo: TestInfo) => {
```

Comment:
This should probably be a common function that all of our tests can use. You did something similar in errors spec

Reply:
Yeah, I actually have done something similar in almost all the specs that I wrote at this point 😄 I'll refactor it.

Reply:
I'm going to introduce a new function for this. But I'll do the refactoring of existing functions as part of another PR that I'm planning to do for test improvements.